core/templates/dev/head/domain/statistics/FeedbackImprovementTaskObjectFactory.ts

```diff
@@ -81,6 +81,11 @@ angular.module('oppia').factory('FeedbackImprovementTaskObjectFactory', [
       return this._actionButtons;
     };
 
+    /** @returns {number} - The time that the task was last updated. */
+    FeedbackImprovementTask.prototype.getLastUpdatedTime = function() {
+      return this._feedbackThread.last_updated;
```

Comment:
Something's a bit off here -- I would expect it to be lastUpdated instead. 

It would extend the scope of this PR a little bit, but would you mind changing the constructor above to define this._status, this._subject, this._lastUpdated, etc. instead of just copying the backend dict wholesale (which is meant to be just a data transfer object)?

Reply:
OK will take a look later today.

Reply:
Currently there is this prototype function:
```
    /**
     * @returns {string} - Data required by the associated directive for
     *    rendering.
     */
    FeedbackImprovementTask.prototype.getDirectiveData = function() {
      return this._feedbackThread;
    };

```
which is used in improvements-tab.directive.html (I've not yet looked into exactly how it is used). So the this._feedbackThread member appears to be necessary. Given that, then surely the DRY thing to do is use the .status, .subject etc. members from this._feedbackThread rather than also defining this._status etc.?
I'll go and study how .getDirectiveData() is used and see how necessary it is.

Reply:
Sounds good, I think looking at .getDirectiveData() is a good idea.

Ah -- I might have been unclear. Note that the data will still exist, just as separate args. The idea is to get rid of the this._feedbackThread dict altogether and replace it with separate args so that we don't have an opaque blob that's being passed around. This won't violate DRY because there isn't repetition of logic (but I think I was unclear because I didn't explicitly say that this._feedbackThread should be deleted). Does that help clarify?

Reply:
Hi @Seanlip. I'm afraid this extra comment is just because I guess the polite way to attract someone's attention is to @ mention them, which I didn't do in my last couple of replies to you.

I'm just trying to follow the code that uses `.getDirectiveData()` and although I am not yet confident in understanding Angular/TypeScript code I reckon it is being used. Am I right that it's what `$scope.getData()` is referring to in `feedback-improvement-task.directive.ts`?

Reply:
Yes, I think you're right (re your last question)!

So I think this boils down to figuring out what info is actually required by $scope.getData(), and modifying the calls to pass that info as separate params. Alternatively, if the info required is this entire domain object then maybe we should pass that instead. But we shouldn't be passing in an opaque dictionary blob (like what's being done now).

Reply:
Ah. I understand better now. Busy weekend - longer reply Monday.

Reply:
Hi @seanlip. I am happy to investigate this. However your suggestion earlier that this be a separate PR makes sense to me. It's a different problem.
My slight ulterior motive is that if that allows this (my first) PR to be accepted then I can more legitimately say 'I contribute to an open source project'. ;-)

Reply:
FYI I filed #8277. Please feel free to leave a comment on it if you'd like to take it up while waiting for @brianrodri's review!

Reply:
OK.